### PR TITLE
Expore pAddon function

### PR DESCRIPTION
**Pull request by alexflorisca**

### CODE REVIEW

File: src/js/permutive.js [hunk 1]
```diff
@@ -109,6 +108,17 @@ class Permutive {
...
 		return Array.from(rootEl.querySelectorAll('[data-o-component="o-permutive"]'), rootEl => new Permutive(rootEl, opts));
 	}
+
+	/**
+	 * Add data to Permutive
+	 * @param {Object} userDemog
+	 * @param {Object} pageMeta
+	 */
+	static pAddon(userDemog, pageMeta) {
+		let user = { "user": Object.assign(userDemog) };
```
Comment (SamiTriki) [inline src/js/permutive.js]: I find these assign a bit hard to follow, do you think it could be clearer with a spread syntax?

Comment (alexflorisca) [reply src/js/permutive.js]: Yeah I agree, but I'll change that in the next PR to keep it simple

Comment (carlesandres) [reply src/js/permutive.js]: @alexflorisca Are you trying to avoid mutating `userDemo`?. If so, I believe the correct syntax is:

```suggestion
		let user = { "user": Object.assign({}, userDemog) };
```

File: src/js/permutive.js [hunk 2]
```diff
@@ -109,6 +108,17 @@ class Permutive {
...
+		let data = { "page": Object.assign(pageMeta, user) };
```
Comment (carlesandres) [inline src/js/permutive.js]: And same here:
```suggestion
		let data = { "page": Object.assign({}, pageMeta, user) };
```

Comment (alexflorisca) [reply src/js/permutive.js]: Thanks Carles, you're right. However I didn't write that function, I simply moved it from another file into here in order for it to be exported. So I'm going to take all of these comments and action them in my next PR - tidying up code and tests

File: src/js/permutive.js (hunk 1 above)
Comment (alevito) [reply src/js/permutive.js]: Unless the code gets transpiled by OBT - or we use babel -, it might not be the best idea to use the spread object instead of Object.assign, as some browser might still not support it.

I think that the first Object.assign could be simply removed
```
{ user: Object.assign(userDemog) }
```
is the same as
```
{user: userDemog}
```
, while the second Object.assign would better be rewritten adding an aempty object as first argument, as follow.
```
let data = {page: Object.assign({}, pageMeta, user)}
```
It's just good practice, in order to make sure that pageMeta is not overwritten with attributes coming from the object user
